wagtail/models/__init__.py

```diff
@@ -3489,10 +3489,12 @@ def delete(self, user=None, **kwargs):
         """
         Custom delete handler to aid in logging
         :param user: the user removing the view restriction
-        :param specific_instance: the specific model instance the restriction applies to
         """
         specific_instance = self.page.specific
         if specific_instance:
+            removed_restriction_type = PageViewRestriction.objects.filter(
+                page=self.page
```

Comment:
This code is re-fetching the database copy of the PageViewRestriction that we're currently working with, to find out the previous state of `restriction_type`. Looking it up based on the page object is a bit long-winded, since we already have the PageViewRestriction object's ID and can just fetch based on that:
```suggestion
                id=self.id
```
That way, we don't have to worry about details like "what happens if the page has more than one restriction on it?". (Which shouldn't happen, but if we can avoid thinking about it, we should :-) )